### PR TITLE
docs: add migration instructions to the first lit@2.0 releases

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Add migration instructions for the pending `lit@2.0` release!

## Motivation and context
Docs are great.

## How has this been tested?

-   [ ] _Test case 1_
    1. Go to https://lit-migration--spectrum-web-components.netlify.app/guides/migration-guide/
    2. Read the page.

## Types of changes
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: projects/documentation/content/guides/migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+					<td class="spectrum-Table-cell">
+						<a href="https://www.npmjs.com/package/{{ pkg.name }}/bundle/v/{{ pkg.version }}">{{ pkg.version }}</a>
+					</td>
+				<tr>{% endfor %}
+			</tbody>
+		</table>
+	</div>
+</details>
+
+### Using `lit@2.0` inside of `lit-html` and/or `LitElement`
```
Comment (Westbrook) [inline projects/documentation/content/guides/migration-guide.md:36]: Clarifies the required changes we've seen in test upgrades to date.

File: projects/documentation/content/guides/migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        return html`
+            // ...
+            <sp-icon-redo></sp-icon-redo>
+            // ...
+        `;
+    }
+}
+```
+
+### Javascript re-exports
```
Comment (Westbrook) [inline projects/documentation/content/guides/migration-guide.md:87]: Clarifies some changes that are not yet in #1401 but have been previously announced the the leads team.

File: projects/documentation/content/guides/migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        return html`
+            // ...
+            <sp-icon>${RedoIcon()}</sp-icon>
+            // ...
+        `;
+    }
+}
+```
+
+In this case the `RedoIcon()` literal will be wrapped in the `html` template literal tag as supplied by `@spectrum-web-components/icons-workflow`. After this update, the version of this tag will no longer be compatible with the `html` template literal tag applied in the `render()` method of your custom element. You can correct this by applying your local value for `html` to the icon literals:
```
Comment (hunterloftis) [inline projects/documentation/content/guides/migration-guide.md:57]:
```suggestion
Here, `RedoIcon()` returns a value built with the `html` dependency of `@spectrum-web-components/icons-workflow`. After this update, that value will no longer be compatible with the `html` tag in the `render()` function here, which is still built with the `lit-html` module.

There are two ways to correct this. We recommend replacing template-literal icons with icon elements, resulting in better performance and decoupled dependencies:
```

File: projects/documentation/content/guides/migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+```js
+import { html } from 'lit-html';
+import {
+    RedoIcon,
+    setCustomTemplateLiteralTag,
+} from '@spectrum-web-components/icons-workflow';
+
+setCustomTemplateLiteralTag(html);
+```
```
Comment (hunterloftis) [inline projects/documentation/content/guides/migration-guide.md:67]: If you accept the above suggestion, the examples will need to be flipped.

File: projects/documentation/content/guides/migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+Or, you can avoid running into issues like this in the future, while leveraging more performant icon delivery, by leveraging fully registered icon elements instead:
```
Comment (hunterloftis) [inline projects/documentation/content/guides/migration-guide.md:69]:
```suggestion
Alternatively, provide your local value for `html` to the icon templates via `setCustomTemplateLiteralTag`:
```

File: projects/documentation/content/guides/migration-guide.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+export { repeat } from 'lit/directives/repeat.js';
+export { classMap } from 'lit/directives/class-map.js';
+export type { ClassInfo } from 'lit/directives/class-map.js';
+export { styleMap } from 'lit/directives/style-map.js';
+export type { StyleInfo } from 'lit/directives/style-map.js';
+export { until } from 'lit/directives/until.js';
+export { live } from 'lit/directives/live.js';
+```
+
+Additional `lit` exports can be acquired directly from that package and with the reduction of `instanceof`-centric checking within the `lit-html` library it is possible that overtime we lean towards that being the preferred path towards acquiring this values.
```
Comment (hunterloftis) [inline projects/documentation/content/guides/migration-guide.md:132]:
```suggestion
Additional `lit` exports can be acquired directly from the `lit` package.

With Lit 2.0 bringing same-document library-version flexibility, re-exporting from shared base libraries is less critical. We may eventually recommend direct imports from `lit` as the preferred path towards acquiring these values.
```

**Review by hunterloftis — DISMISSED** [review]
Let's ship it!

I left several copy suggestions, nothing blocking.

**Pull request merged**